docker/Dockerfile.rocm-deps-only

```diff
@@ -0,0 +1,92 @@
+# Docker container for running DGL on AMD GPUs with ROCM. Includes only the
+# dependencies of DGL, not DGL itself.
+
+# TODO: This image is quite large. Consider multistage builds.
+# See also https://github.com/ROCm/ROCm-docker/issues/120
+FROM rocm/pytorch:rocm6.3.3_ubuntu22.04_py3.10_pytorch_release_2.4.0
+
+# Create a venv that is active for all subsequent RUN commands and when the
+# image is run as a container. See
+# https://pythonspeed.com/articles/activate-virtualenv-dockerfile/
+WORKDIR /
+ENV VIRTUAL_ENV=/.venv
+# We use system-site-packages so that we get PyTorch from the base image.
+# Otherwise it's tricky to get a compatible PyTorch version as Torch 2.6 changes
+# the default for torch.load weights_only and ROCm 6.3 is in the nightlies,
+# which doesn't have torch<2.6.
+RUN python3 -m venv --system-site-packages $VIRTUAL_ENV
+ENV PATH="$VIRTUAL_ENV/bin:$PATH"
```

Comment:
You might want to source the `$VIRTUAL_ENV/bin/activate` this time (or run the command mentioned later) because the script does also ensure new path priorities are used with `hash -r 2> /dev/null`. I think it's low risk, but also a nightmare to debug if we get bitten by it.

Disclaimer: I'm just going on the comment for that command. I don't understand what it does.

Reply:
Hmm interesting point. My usage here is based on the linked post and my own reading of the activate script. I'm pretty sure the bash command hash is reset per session and each invocation of `RUN` in a dockerfile is a new bash session (this is in fact why sourcing the venv activate script in the dockerfile doesn't work). Also, I think it's reset if `PATH` is modified, but my only reference for that is https://unix.stackexchange.com/a/86017. I just tested and both of those are true in my own bash. It's interesting though because the latter finding is contradicted by the venv comment.

Reply:
Then it sounds safe to me and like we don't need it